standards/ICRC-3/ICRC-3.did

```diff
@@ -0,0 +1,101 @@
+type TxIndex = nat;
+
+type Account = record {
+    owner : principal;
+    subaccount : opt blob;
+};
+
+type GetTransactionsRequest = record {
+    // The index of the first tx to fetch.
+    start : TxIndex;
+    // The number of transactions to fetch.
+    length : nat;
+};
+
+type GetTransactionsResponse = record {
+    // The total number of transactions in the log.
+    log_length : nat;
+
+    // List of transaction that were available in the ledger when it processed the call.
+    //
+    // The transactions form a contiguous range, with the first transaction having index
+    // [first_index] (see below), and the last transaction having index
+    // [first_index] + len(transactions) - 1.
+    //
+    // The transaction range can be an arbitrary sub-range of the originally requested range.
+    transactions : vec Transaction;
+
+    // The index of the first transaction in [transactions].
+    // If the transaction vector is empty, the exact value of this field is not specified.
+    first_index : TxIndex;
+
+    // Encoding of instructions for fetching archived transactions whose indices fall into the
+    // requested range.
+    //
+    // For each entry `e` in [archived_transactions], `[e.from, e.from + len)` is a sub-range
+    // of the originally requested transaction range.
+    archived_transactions : vec record {
```

Comment:
The `archived_transactions` field feels a bit unrelated to me. I would expect this as a response of a meta data request. Or is this related to the specified requested range in some way?

Reply:
> Or is this related to the specified requested range in some way?

Yes, the ledger restricts entries in the archived transactions list to the range the client requested.

We didn't want to have a separate "metadata" endpoint that returns a range -> canister assignment because this API would suffer from race conditions: by the time you received the response, the assignment of the tail might have moved already.